### PR TITLE
Add @types/map-stream

### PR DESCRIPTION
**Pull request by KonradPerlicki**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/map-stream/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+         * Default: `false`
+         */
+        failures?: boolean;
+    }
+
+    type Callback = (err: null | Error, data: unknown) => void;
+}
+
+declare function mapStream(
+    mapper: (data: unknown, callback: mapStream.Callback) => unknown,
```
Comment (peterblazejewicz) [inline types/map-stream/index.d.ts:24]: does it return unknown or just void (ev undefined):
https://github.com/dominictarr/map-stream/blob/master/index.js#L83-L85

File: types/map-stream/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export = mapStream;
+
+declare namespace mapStream {
+    interface Options {
+        /**
+         * Continue mapping even if error occured.
+         *
+         * On error `map-stream` will emit `failure` event.
+         *
+         * Default: `false`
```
Comment (peterblazejewicz) [inline types/map-stream/index.d.ts:15]:
```suggestion
         * @default false
```

File: types/map-stream/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        failures?: boolean;
+    }
+
+    type Callback = (err: null | Error, data: unknown) => void;
+}
+
+declare function mapStream(
+    mapper: (data: unknown, callback: mapStream.Callback) => unknown,
+    opts?: mapStream.Options,
+): unknown;
```
Comment (peterblazejewicz) [inline types/map-stream/index.d.ts:26]: it does return instance of a `Stream`, does it?
(adding this type would require Node types refererence)

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
Please handle feedback

File: types/map-stream/index.d.ts (hunk 3 above)
Comment (KonradPerlicki) [reply types/map-stream/index.d.ts]: It does, I fixed it

File: types/map-stream/index.d.ts (hunk 1 above)
Comment (KonradPerlicki) [reply types/map-stream/index.d.ts]: It seems to me that it doesn't return anything

**Review by sheetalkamat — APPROVED** [review]

File: types/map-stream/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/map-stream/index.d.ts]: cool, so just type `void` (so no one is expected to consume return results)

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@KonradPerlicki thanks!

**Comment by KonradPerlicki** [comment]
Ready to merge

**Pull request merged**